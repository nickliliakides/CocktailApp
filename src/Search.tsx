import React, { FC } from 'react';

interface IProps {
  handleSubmit: (e: React.FormEvent<HTMLElement>) => void
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const Search: FC<IProps> = ({ handleSubmit, handleChange }): JSX.Element => {
  return (
    <form className='form' onSubmit={handleSubmit}>
      <div className='form-group'>
        <label htmlFor='search'>Find a cocktail by name</label>
        <div className='search'>
          <input
            className='form-control'
            id='search'
            name='search'
            type='search'
            placeholder='Please enter a cocktail name...'
            onChange={handleChange}
          />
          <button type='submit' className='btn btn-primary btn-lg'>
            Search
          </button>
        </div>
      </div>
    </form>
  );
};

export default Search;
