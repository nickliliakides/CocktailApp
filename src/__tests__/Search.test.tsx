import { shallow, ShallowWrapper } from 'enzyme';
import Search from '../Search';

let wrapper: ShallowWrapper;

const defaultProps = {
  handleSubmit: jest.fn(),
  handleChange: jest.fn(),
};

describe('<Search /> component', () => {
  beforeEach(() => {
    wrapper = shallow(<Search {...defaultProps} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should matches its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
