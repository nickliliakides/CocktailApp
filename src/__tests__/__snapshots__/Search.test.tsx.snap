// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should matches its snapshot 1`] = `
<form
  className="form"
  onSubmit={[MockFunction]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="search"
    >
      Find a cocktail by name
    </label>
    <div
      className="search"
    >
      <input
        className="form-control"
        id="search"
        name="search"
        onChange={[MockFunction]}
        placeholder="Please enter a cocktail name..."
        type="search"
      />
      <button
        className="btn btn-primary btn-lg"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</form>
`;
